library/Backends/llvm/src/Juvix/Backends/LLVM/Codegen/Types/Shared.hs

```diff
@@ -0,0 +1,34 @@
+-- | Shared between Types and Sum
+module Juvix.Backends.LLVM.Codegen.Types.Shared where
+
+import Juvix.Library hiding (Type)
+import qualified Juvix.Library.HashMap as Map
+import LLVM.AST
+import qualified LLVM.AST as AST ()
+import qualified LLVM.AST.Constant as C ()
+import LLVM.AST.Global as Global ()
+
+type SymbolTable = Map.T Symbol Operand
+
+type TypeTable = Map.T Symbol Type
+
+data SumInfo = S
+  { sum' :: Symbol,
+    offset :: Int,
+    tagSize' :: Word32
+  }
+  deriving (Show, Eq)
+
+-- | a mapping between the variant and the sum type along with
+-- the tag associated with it
+type VariantToType = Map.T Symbol SumInfo
+
+type Names = Map.T Symbol Int
+
+uniqueName :: Symbol -> Names -> (Symbol, Names)
+uniqueName nm ns =
+  case Map.lookup nm ns of
+    Nothing -> (nm, Map.insert nm 1 ns)
+    Just ix -> (intern (show nm <> show ix), Map.insert nm (succ ix) ns)
+
+instance Hashable Name
```

Comment:
How does this instance behave?

Reply:
The instance or the unique name code above it?